### PR TITLE
Add luks encryption

### PR DESCRIPTION
**Pull request by PeterQFR**
# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits follow the conventional commit [specification](https://www.github.com/mendersoftware/mendertesting/commitlint/grammar.md) for the Mender project.

The majority of our contributions are fixes, which means your commit should have
the form below:

fix: Add LUKS Encryption

Attempt to enable Mender to encrypt the rootfs and data partitions to ensure data security at rest. 

Changelog: 
Ticket:``

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

This is a partial solution to change the mender-convert script to firstly place the rootfsA and B and data in an Encrypted LUKS volume as LVM2 volumes. I am still working out how mender selects the rootfs partition, I did change the /etc/mender/mender.conf file to include the volume paths as 
```
"rootfsPartA": "/dev/<volume_group_name>/<volume_name",
"rootfsPartB": "/dev/<volume_group_name>/<volume_name"
```
which is similar to what ubuntu kernel command line use as the `root=` parameter, though I think it is just the volume name.

However after copying the produced deployable image it doesn't get past grub. I think there are some more steps required which is why I have produced this pull request to get help modifying the boot sequence to support LUKS encryption.

It is based off a conversation [here](https://hub.mender.io/t/encryption-of-rootfs-a-b-partitions-unattended-using-tpm/5710)

### CODE REVIEW

File: mender-convert-package [hunk 1]
```diff
@@ -277,82 +279,151 @@ else
...
+  # Keep the uuid of the original gpt table for the boot partition
+  boot_partuuid=$(disk_get_part_value "${disk_image}" 1 UUID)
+  log_info "Updating GPT partition uuid of the boot partition in image: ${img_path} to ${boot_partuuid}"
+  run_and_log_cmd "sgdisk -u ${MENDER_BOOT_PART_NUMBER}:${boot_partuuid} '${img_path}'"
+  
+  run_and_log_cmd "${PARTED} -s ${img_path} set 1 boot on"
+  run_and_log_cmd "${PARTED} -s ${img_path} -- unit s mkpart primary ext4 ${rootfsa_start} ${data_end}"
+  
+  #TODO Determine if using LUKS logical partitions does something to the size required.  
+  loop_device=$(losetup --offset $(( ${rootfsa_start} * 512 )) --sizelimit $((( 2 * ${rootfs_part_sectors} + ${data_part_sectors} * 2 ) * 512 )) --show --find ${img_path})
```
Comment (PeterQFR) [inline mender-convert-package:297]: Note that I didn't get loop device access working on docker, in testing I did this on a sacrificial linux box.

**Comment by PeterQFR** [comment]
is there some override on the mender.conf `rootfsPartA` variables? When I look through it seems that the system has overriden them in mender-conver-modify, eg in the build output:
```
....
Creating state folder in the data partition for Mender add-ons
2023-04-10 20:41:54 [INFO] [mender-convert-modify] Not installing GRUB EFI bootloader, relying on platform provided one.
2023-04-10 20:41:54 [INFO] [mender-convert-modify] Installing Mender GRUB tools...
2023-04-10 20:41:54 [INFO] [mender-convert-modify] Installing the GRUB editenv binary
2023-04-10 20:41:54 [INFO] [mender-convert-modify] Generating grub config using update-grub...
2023-04-10 20:42:01 [INFO] [mender-convert-modify] Using root device A in mender.conf: /dev/sda2
2023-04-10 20:42:01 [INFO] [mender-convert-modify] Using root device B in mender.conf: /dev/sda3
2023-04-10 20:42:01 [INFO] [mender-convert-modify] Creating state scripts version file.
2023-04-10 20:42:01 [INFO] [mender-convert-modify] Installing a custom /etc/fstab (see /home/peter/mender-drive/mender-convert/logs/convert.log for more info)
....
```

File: mender-convert-package (hunk 1 above)
Comment (kacf) [reply mender-convert-package:297]: Hmm, it should have the privileges to use loop devices. What error do you get?

**Comment by kacf** [comment]
> is there some override on the mender.conf `rootfsPartA` variables?

I think they are set from `MENDER_STORAGE_DEVICE_BASE` and `MENDER_ROOTFS_PART_A_NUMBER`. I do not know off the top of my head if you can set `MENDER_ROOTFS_PART_A` to an explicit or not, or if it will always be overwritten. I would check out the part around [here](https://github.com/mendersoftware/mender-convert/blob/master/modules/disk.sh#L271), I think that's where the magic happens.

**Pull request closed without merging**